aws/resource_aws_glue_partition.go

```diff
@@ -0,0 +1,360 @@
+								},
+							},
+						},
+						"skewed_info": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"skewed_column_names": {
+										Type:     schema.TypeList,
+										Optional: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+									},
+									"skewed_column_values": {
+										Type:     schema.TypeList,
+										Optional: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+									},
+									"skewed_column_value_location_maps": {
+										Type:     schema.TypeMap,
+										Optional: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+									},
+								},
+							},
+						},
+						"sort_columns": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"column": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"sort_order": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+								},
+							},
+						},
+						"stored_as_sub_directories": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"parameters": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"creation_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_analyzed_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_accessed_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func readAwsGluePartitionID(id string) (catalogID string, dbName string, tableName string, values []string, error error) {
+	idParts := strings.Split(id, ":")
+	if len(idParts) != 4 {
+		return "", "", "", []string{}, fmt.Errorf("expected ID in format catalog-id:database-name:table-name:values, received: %s", id)
+	}
+	vals := strings.Split(idParts[3], "#")
+	return idParts[0], idParts[1], idParts[2], vals, nil
+}
+
+func resourceAwsGluePartitionCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).glueconn
+	catalogID := createAwsGlueCatalogID(d, meta.(*AWSClient).accountid)
+	dbName := d.Get("database_name").(string)
+	tableName := d.Get("table_name").(string)
+	values := d.Get("partition_values").(*schema.Set)
+
+	input := &glue.CreatePartitionInput{
+		CatalogId:      aws.String(catalogID),
+		DatabaseName:   aws.String(dbName),
+		TableName:      aws.String(tableName),
+		PartitionInput: expandGluePartitionInput(d),
+	}
+
+	log.Printf("[DEBUG] Creating Glue Partition: %#v", input)
+	_, err := conn.CreatePartition(input)
+	if err != nil {
+		return fmt.Errorf("error creating Glue Partition: %w", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s:%s:%s:%s", catalogID, dbName, tableName, stringifyAwsGluePartition(values)))
```

Comment:
Maybe add a _create Glue partition ID_ function to parallel `readAwsGluePartitionID`.
There is also now precedent in moving these ID create/read functions to an `internal/service` package.
See e.g. `aws/internal/service/ec2/id.go`.

Reply:
done.